### DISCUSSION TITLE
Fake login: Local VS api

### DISCUSSION THREAD

**Discussion by TienPham18** [neuefische/questions#752] [Web] [answered]
We have spent our day attempting to implement a user login (by following [Diego´s example](https://codesandbox.io/p/sandbox/next-fake-login-example-j3tr7s?file=%2FREADME.mdhttps://codesandbox.io/p/sandbox/next-fake-login-example-j3tr7s?file=%2FREADME.md)). We are, however, having trouble making the change in the code from local solution to a MongoDB solution. 

The code that seems to be causing us the most trouble is line 3 in the following:

```
export default function handler(request, response) {
  if (request.method === "POST") {
    const user = findByEmail(request.body.email);
    if (!user) {
      response.status(400).json({ message: "Incorrect Credentials" });
      return;
    }
    if (request.body.roomNumber !== roomNumber) {
      response.status(400).json({ message: "Incorrect Credentials" });
      return;
    }
    response.status(200).json(user);
    return;
  }
  response.status(405).json({ message: "Method Not Allowed" });
}
```

This code is from `pages/api/users/login.js`.

We have looked through all of the Backend Handouts and tried several different things, none of which have worked. We are afraid the solution is super-easy and staring us in the face, but we can´t see it. 

Here is the link to the [PR](https://github.com/ohlunlacassi/hotel-tishjama/pull/29).

Thank you very much,
Hotel Tishjama

**Comment by moonwave99** [comment]
The `findByEmail` function was defined just in the example - since you have a `User` collection in mongo, you can use `User.findOne` like:

```js
import User from "@/db/models/User.js";

export default function handler(request, response) {
  if (request.method === "POST") {
    const user = User.findOne({ email: request.body.email });
    if (!user) {
      response.status(400).json({ message: "Incorrect Credentials" });
      return;
    }
    // remember to check against user.roomNumber!
    if (request.body.roomNumber !== user.roomNumber) {
      response.status(400).json({ message: "Incorrect Credentials" });
      return;
    }
    response.status(200).json(user);
    return;
  }
  response.status(405).json({ message: "Method Not Allowed" });
}
```

Let us know if it works now!

**Reply by shrocket** [reply]
It does not work now. 😢 
We must be missing something somewhere else ... sorry!
We keep getting an "Incorrect Credentials" error, but we've checked and double-checked our input against MongoDB, and we're not typing anything wrong. 🤷 
We will now compare your example sandbox to our code again.

**Reply by shrocket** [reply]
We also thought it might have something to do with the fact that there is a fetcher on both `_app.js` and `hooks/useUser.js`, so we deleted both fetchers, wrote a fetcher in /lib and imported that fetcher to the aforementioned pages. It didn't work. 😢

**Reply by moonwave99** [reply]
Log `user` and `request.body` in the server, and see if you like what you see inside or not!

Do all key name match? Data types as well (numbers vs strings), etc...

**Comment by shrocket** [comment, marked as answer]
Thanks to @JessicaLoers, we have solved the problem! It had to do with connecting to the database, and then with an async/await issue . . .

**Comment by TienPham18** [comment]
Thanks to @JessicaLoers one more time!
